### ISSUE TITLE
SBC install appimage arm64

### ISSUE THREAD

**Issue by makerorne** [sourceryltd/makerhub-releases#6]
Hello and thanks for the new Makerhub !!!

i want to use Makerhub on a SBC (rock pi 3c) because it can control spindle on/off with GPIO (as with 1.1.2 version of makerverse)
I tried to install on debian 11 (xfce):
first, i had problem installing arm64 appimage file: a pb with libz.so:
i solved it with this command:
sudo apr-get install -y zlib1g-dev

but now i have this problem.... (see in file)
thanks for help
![PXL_20240329_090244226](https://github.com/sourceryltd/makerhub-releases/assets/79270587/209e1356-6aa2-4e24-baba-2120f90bf26e)

**Comment by kevinjohnson123** [comment]
I recently had a similar issue on another ticket.  

https://github.com/sourceryltd/makerhub-releases/issues/7

Try downloading the    ARM v7L   AppImage from the releases section.  You might need to install fuse2 or fuse3 in addition.  Depends on how updated the Rasapian OS system is.

**Comment by shiftylilbastrd** [comment]
Edit: This is an RPI4
OS is fully updated and I already had both fuse2 and fuse3 installed, the other two return
 ```
E: Unable to locate package fuse-utils
E: Unable to locate package python-fuse
```
And trying to run the appimage returns 
```
bash: ./MakerHub-1.3.1-beta1-armv7l.AppImage: cannot execute: required file not found
```

**Comment by kevinjohnson123** [comment]
i probably need to build directly on  Pi4 hardware then.   It's interesting that it says a file is missing but doesn't tell you what file it can't find.   Just curious what   dmesg logs show after you attempt to load the AppImage...

**Comment by shiftylilbastrd** [comment]
Any progress here? There's a Pi 5 now and we can't even use a 4.

**Comment by kevinjohnson123** [comment]
the problem is that one of the  libraries used in MakerHub/Makerverse will not build on any other architecture than  ARM v7.  MakerMade has an M4-like machine that is planned to be launched around September which does not require MakerHub at all.   We can spend resources that are incredibly tight on software that will be obsolete in 3-4 months,  or use those resources on the new machine.
Technically speaking i don't currently work for MakerMade anymore, but if there is enough people with M2 machines that absolutely must get this software on a  Pi4 or Pi5 maybe a bug bounty?  Until then the way you fix this is to buy a  Raspberry Pi3 B+, they're still manufactured oddly enough, and they're cheap.

**Comment by makerorne** [comment]
so; on a pi 3B+ it will run without problem ?

**Comment by kevinjohnson123** [comment]
> so; on a pi 3B+ it will run without problem ?

yes, it was built on a Pi 3B+  using  PiOS.  The real problem seems to be with the Serial Port Library and version conflicts with other NodeJS  libraries in the software.   I can upgrade the serial port library to a version that will build on  a Pi4 and evne Pi5, but that breaks  calls to those other NodeJS libraries.   It's basically  a  "dependency hell" problem.    So for the ARM version of MakerHub,  we're really limited to  ARMv7a  architecture.
If for some reason you can't get a  Pi3, don't want to use a Pi3,  the  Linux version of the software works perfectly fine on anything that runs intel architecture.  I haven't tested it specifically on any of the  Intel based  SBCs out there, but as long as you can install Ubuntu or Debian to those boards,  you just run the   .DEB installer file  and you're up and running.

**Comment by makerorne** [comment]
thanks for your answer
**So i will try on an old pi3b+**

and for me;, i dont want to use a computer (linux or windows) because i use the output on the GPIO to start an stop the router with functions on makerverse.
I am really surprised that nobody try to command the router....
and in my mind, i would like to control spindle speed with the 0-10 v but for the moment i don't know how to...

Or another way should be to send http command to switch a smartplug.... but i don't know how to do this...
